pkg/controllers/spannerautoscaler_controller_test.go

```diff
@@ -409,7 +410,164 @@ func Test_calcDesiredNodes(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			if got := calcDesiredNodes(tt.args.currentCPU, tt.args.currentNodes, tt.args.targetCPU, tt.args.minNodes, tt.args.maxNodes, tt.args.maxScaleDownNodes); got != tt.want {
-				t.Errorf("calcDesiredNodes() = %v, want %v", got, tt.want)
+				t.Errorf("calcDesiredProcessingUnits() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_calcDesiredProcessingUnits(t *testing.T) {
+	type args struct {
+		currentCPU             int32
+		currentProcessingUnits int32
+		targetCPU              int32
+		minProcessingUnits     int32
+		maxProcessingUnits     int32
+		maxScaleDownNodes      int32
+	}
+	tests := []struct {
+		name string
+		args args
+		want int32
+	}{
+		{
+			name: "scale up",
+			args: args{
+				currentCPU:             50,
+				currentProcessingUnits: 300,
+				targetCPU:              30,
+				minProcessingUnits:     100,
+				maxProcessingUnits:     1000,
+				maxScaleDownNodes:      2,
+			},
+			want: 600,
```

Comment:
Shouldn't this be 500? Calculating as follows:

`desiredPU` = (`currentCPU`  * `currentPU` ) / `targetCPU`

500 = ( 50 * 300 ) / 30

Reply:
I think we should align with the current behavior in nodes.
https://github.com/mercari/spanner-autoscaler/blob/ab62d5b4981e7fc2dc0917348a0160e12cc5f062/pkg/controllers/spannerautoscaler_controller_test.go#L347-L358

To explain the current behavior in words, it tries to keep the CPU utilization below the Target CPU, not "or equal".